### DISCUSSION TITLE
Is it possible to create and use a custom provider

### DISCUSSION THREAD

**Discussion by sbrunner** [geopython/pygeoapi#1229] [Q&A] [answered]
I'm looking to implement the OGC API Tiles to our tilling solution: https://github.com/camptocamp/tilecloud-chain/
And I'm having a look at this project if he can solve the major part of the requirements.

Then my question is: It is possible to build an application based on this project that uses a custom provider?

Thanks in advance

**Comment by justb4** [comment, marked as answer]
Yes, see the documentation: https://docs.pygeoapi.io/en/latest/plugins.html. You can create a Provider as a plugin and have pygeoapi instantiating it. In the `name` attribute of the Provider YAML config you should point to your Python module/class which should be in the PYTHONPATH, or at least be able to be found. There are several examples in the docs, pointing to external repos like https://github.com/eurodatacube/ogc-edc/tree/oapi/edc_ogc/pygeoapi.
